radixdlt/src/main/java/com/radixdlt/consensus/AddressBookValidatorSetProvider.java

```diff
@@ -59,15 +69,35 @@ public AddressBookValidatorSetProvider(ECPublicKey selfKey, AddressBook addressB
 			).distinct().collect(Collectors.toList()))
 			.filter(peers -> peers.size() == fixedNodeCount)
 			.firstOrError()
-			.map(peers -> ValidatorSet.from(
+			.map(peers ->
 				peers.stream()
+					.sorted(Comparator.comparing(ECPublicKey::euid))
 					.map(p -> Validator.from(p, UInt256.ONE))
-					.collect(Collectors.toList())
-			))
+					.collect(ImmutableList.toImmutableList())
+			)
 			.cache();
 	}
 
 	public ValidatorSet getValidatorSet(long epoch) {
-		return validatorSet.blockingGet();
+		ImmutableList<Validator> validators = validatorList.blockingGet();
+
+		Builder<Validator> validatorSetBuilder = ImmutableList.builder();
+		Random random = new Random(epoch);
+		List<Integer> indices = IntStream.range(0, validators.size()).boxed().collect(Collectors.toList());
+		for (long i = 0; i < epoch; i++) {
```

Comment:
I suppose this is intended to ensure consistent outputs for the same epoch? A few comments would help here, given how critical this is to the security of the network.